src/Makefile.housekeeping

```diff
@@ -1187,7 +1186,7 @@ $(BIN)/version.%.o : core/version.c $(MAKEDEPS) $(GIT_INDEX)
 $(BIN)/%.tmp : $(BIN)/version.%.o $(BLIB) $(MAKEDEPS) $(LDSCRIPT)
 	$(QM)$(ECHO) "  [LD] $@"
 	$(Q)$(LD) $(LDFLAGS) -T $(LDSCRIPT) $(TGT_LD_FLAGS) $< $(BLIB) -o $@ \
-		--defsym _build_id=`$(BUILD_ID_CMD)` \
+		--defsym _build_id=$(shell sha1sum $^|sort|cksum|awk '{printf "0x%08x", $$1}') \
```

Comment:
Fix this to avoid exceeding 80 characters, and I'm happy to merge this :)

Reply:
I'll fix this too before resubmitting updates.

Reply:
While this is very similar to my #82, there is a difference: the `sha1sum $^` will include the filenames in output - if those are absolute pathnames(are they?), it would mean that a binary built in `/home/bernhard` would be different from yours built from identical sources.
and for some reason I still get 4 random bytes in the binaries - maybe because openSUSE's `.o` files are from LTO and gcc's intermediate representations have randomness. I'll retest without LTO.

Reply:
> While this is very similar to my #82, there is a difference: the `sha1sum $^` will include the filenames in output - if those are absolute pathnames(are they?), it would mean that a binary built in `/home/bernhard` would be different from yours built from identical sources.

Good point.  @mmlb could you please find a way to omit filenames from the resulting hash value?

Reply:
Ooops didn't see this until now. Let me check on this. I don't think they're absolute filenames though.

Reply:
Nope these are all relative paths. I checked by running trying out:

```diff
diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index 83665f83..2414fbcf 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -1201,6 +1201,7 @@ $(BIN)/version.%.o : core/version.c $(MAKEDEPS) $(GIT_INDEX)
 #
 $(BIN)/%.tmp : $(BIN)/version.%.o $(BLIB) $(MAKEDEPS) $(LDSCRIPT)
 	$(QM)$(ECHO) "  [LD] $@"
+	sha1sum $^| sort | cksum | awk '{printf "0x%08x", $$1}'
 	$(Q)$(LD) $(LDFLAGS) -T $(LDSCRIPT) $(TGT_LD_FLAGS) $< $(BLIB) -o $@ \
 		--defsym _build_id=$(shell $(BUILD_ID_CMD)) \
 		--defsym _build_timestamp=$(BUILD_TIMESTAMP) \
 
```
and running `make bin-x86_64-efi/ipxe.efi` (in src/) and from my home doing `make -C /home/manny/cloned/ipxe/ipxe/src/ bin-x86_64-efi/ipxe.efi` which gave me:

```
make: Entering directory '/home/manny/cloned/ipxe/ipxe/src'
make: /bin/echo: No such file or directory
make: /bin/echo: No such file or directory
  [VERSION] bin-x86_64-efi/version.ipxe.efi.o
  [LD] bin-x86_64-efi/ipxe.efi.tmp
sha1sum bin-x86_64-efi/version.ipxe.efi.o bin-x86_64-efi/blib.a Makefile Makefile.housekeeping .echocheck arch/x86_64/Makefile arch/x86/Makefile arch/x86_64/Makefile.efi arch/x86/Makefile.efi Makefile.efi bin-x86_64-efi/.fnrec.state scripts/efi.lds| sort | cksum | awk '{printf "0x%08x", $1}'
0x54378e64  [FINISH] bin-x86_64-efi/ipxe.efi
rm bin-x86_64-efi/version.ipxe.efi.o
make: Leaving directory '/home/manny/cloned/ipxe/ipxe/src'
```